server/core/static-files.dylan

```diff
@@ -230,7 +230,11 @@ define method copy-to-end
   let buffer :: <sequence> = make(stream-sequence-class(in-stream),
                                   size: buffer-size);
   iterate loop ()
-    let count = read-into!(in-stream, buffer-size, buffer, on-end-of-stream: #f);
+    let count = block ()
+                  read-into!(in-stream, buffer-size, buffer)
+                exception (ex :: <incomplete-read-error>)
+                   ex.stream-error-count
```

Comment:
hrm, there was something fishy about this because at first I had called `ex.incomplete-read-error-count` and it seemed to compile and run, but then later I got a compiler warning that `incomplete-read-error-count` didn't exist so I found `stream-error-count`. I'll take a closer look at the whole thing tonight.

I don't like this API for `read-into!`. Something like this seems simpler / more intuitive maybe?

```dylan
let (count, eof?) = read-into!(...); // no on-end-of-stream parameter at all
```

Reply:
And while I'm bike-shedding the API, I notice that virtually all the callers of `read-into!` (I didn't check every single one) pass `buffer.size` as the second (number of elements to read) argument. Maybe we don't need that at all. It can be calculated as `buffer.size - start`.

Hard to think of an example where you pass a big buffer and don't want to fill it up if you can.

Reply:
ok, took another look. `stream-error-count` is the correct accessor and it contains the number of elements actually read, so I believe my implementation is correct here.

I was probably doing three things at once when I thought that `incomplete-read-error-count` was a thing.

I'll address the docs bug separately. And maybe the API issue, someday.